cmd/buildah/bud.go

```diff
@@ -309,75 +315,120 @@ func budCmd(c *cobra.Command, inputArgs []string, iopts budOptions) error {
-		BlobDirectory:           iopts.BlobCache,
-		CNIConfigDir:            iopts.CNIConfigDir,
-		CNIPluginPath:           iopts.CNIPlugInPath,
-		CommonBuildOpts:         commonOpts,
-		Compression:             compression,
-		ConfigureNetwork:        networkPolicy,
-		ContextDirectory:        contextDir,
-		DefaultMountsFilePath:   globalFlagResults.DefaultMountsFile,
-		Devices:                 iopts.Devices,
-		DropCapabilities:        iopts.CapDrop,
-		Err:                     stderr,
-		ForceRmIntermediateCtrs: iopts.ForceRm,
-		From:                    iopts.From,
-		IDMappingOptions:        idmappingOptions,
-		IIDFile:                 iopts.Iidfile,
-		In:                      stdin,
-		Isolation:               isolation,
-		Labels:                  iopts.Label,
-		Layers:                  layers,
-		LogRusage:               iopts.LogRusage,
-		Manifest:                iopts.Manifest,
-		MaxPullPushRetries:      maxPullPushRetries,
-		NamespaceOptions:        namespaceOptions,
-		NoCache:                 iopts.NoCache,
-		OS:                      imageOS,
-		Out:                     stdout,
-		Output:                  output,
-		OutputFormat:            format,
-		PullPolicy:              pullPolicy,
-		PullPushRetryDelay:      pullPushRetryDelay,
-		Quiet:                   iopts.Quiet,
-		RemoveIntermediateCtrs:  iopts.Rm,
-		ReportWriter:            reporter,
-		Runtime:                 iopts.Runtime,
-		RuntimeArgs:             runtimeFlags,
-		RusageLogFile:           iopts.RusageLogFile,
-		SignBy:                  iopts.SignBy,
-		SignaturePolicyPath:     iopts.SignaturePolicy,
-		Squash:                  iopts.Squash,
-		SystemContext:           systemContext,
-		Target:                  iopts.Target,
-		TransientMounts:         iopts.Volumes,
-		OciDecryptConfig:        decConfig,
-		Jobs:                    &iopts.Jobs,
-	}
-	if iopts.IgnoreFile != "" {
-		excludes, err := parseDockerignore(iopts.IgnoreFile)
+	var (
+		jobSemaphore *semaphore.Weighted
+		builds       multierror.Group
+	)
+	if iopts.Jobs > 0 {
+		jobSemaphore = semaphore.NewWeighted(int64(iopts.Jobs))
+		iopts.Jobs = 0
+	}
+	if iopts.Manifest != "" {
+		// Ensure that the list's ID is known before we spawn off any
+		// goroutines that'll want to modify it, so that they don't
+		// race and create two lists, one of which will rapidly become
+		// ignored.
+		rt, err := libimage.RuntimeFromStore(store, nil)
+		if err != nil {
+			return err
+		}
+		_, err = rt.LookupManifestList(iopts.Manifest)
+		if err != nil && errors.Cause(err) == storage.ErrImageUnknown {
+			list := manifests.Create()
+			_, err = list.SaveToImage(store, "", []string{iopts.Manifest}, manifest.DockerV2ListMediaType)
+		}
 		if err != nil {
 			return err
 		}
-		options.Excludes = excludes
-	}
-	if c.Flag("timestamp").Changed {
-		timestamp := time.Unix(iopts.Timestamp, 0).UTC()
-		options.Timestamp = &timestamp
 	}
+	for _, platform := range platforms {
+		func() {
```

Comment:
What does this closure do?

Reply:
At the time, I think it was for ensuring that each goroutine we ran with builds.Go() was getting the `platform` value from the iteration of the loop that launched it, since otherwise the iterator value would change because it was being shared.  But I don't think we need to be creating a new scope just for that since we're already creating a new variable for each iteration.  Removing it.